cms/app_base.py

```diff
@@ -32,3 +43,17 @@ def get_config(self, namespace):
 
     def get_config_add_url(self):
         raise NotImplemented('Configurable AppHooks must implement this method')
+
+    @property
+    def legacy_urls(self):
+        warnings.warn('Accessing CMSApp.urls directly is deprecated, '
+                      'and it will be removed in version 3.5; CMSApp.get_urls method',
+                      DeprecationWarning)
+        return self._urls
+
+    @legacy_urls.setter
+    def urls(self, value):
+        self._urls = value
+
+    def get_urls(self, page=None):
```

Comment:
Since this is all about flexibility, perhaps it makes sense to include `**kwargs` here?